### ISSUE TITLE
FireFox v60+ compatibility broken

### ISSUE THREAD

**Issue by xGSTx** [fuyu/alexa-rank-webextension#2]
Hi,

This great Addon has stopped working with FireFox v60+ 

No Alexa Information is anywhere to be found!

The Addon itself is no longer clickable in the options button in the url bar.

Please fix this. I cannot live without this Addon. 

Thank You.

![12121](https://user-images.githubusercontent.com/33794631/40578857-0f7ea786-611c-11e8-985e-110852a0f03a.png)

**Comment by fuyu** [comment]
I would fix it if I knew how to reproduce the issue. I'll need more details.

**Comment by xGSTx** [comment]
Well, it just suddenly stopped working after the FireFox automatic update to v60.

I uninstalled and reinstalled your alexa extension but had no luck.

This is all the information i have. I can't tell you what the issue is.

Is there something specific you need?

EDIT: I am using Firefox v60.0.1 - 64-Bit - German Language.

Maybe you can reproduce the Issue with my setup?

**Comment by fuyu** [comment]
Okay, I have an idea how you can get some helpful information. In address bar type `about:debugging` then enable debugging mode, scroll to the extension and click Debug. You should see a console where hopefully something useful could be. But as you can see in my case everything is working:

<img width="1012" alt="screen shot 2018-06-15 at 20 22 33" src="https://user-images.githubusercontent.com/1557716/41481349-37a584be-70da-11e8-9f35-6edf7af6f66b.png">

**Comment by xGSTx** [comment]
Hi,

I did what you said and i got the ouput shown below. I don't have any extension called adblocker or human-web or anti-phishing. This seems to be causing the issue. I use uBlock Origin but that is nowhere to be found in the ouput. 

![fuyu](https://user-images.githubusercontent.com/33794631/41486092-67225b6e-70e3-11e8-80bc-de7c007a4664.png)

I hope this helps. Let me know if there is something else i can do. I don't seem to be the only one having this issue. There are other people with the same problem. I cannot live without this great extension. Please help me fix this.

Regards.

**Comment by fuyu** [comment]
Unfortunately, it doesn't make much sense to me.
Can you make a raw request to alexa.com? Try running in console:
```
$ curl "http://xml.alexa.com/data?cli=10&dat=nsa&url=github.com"
```

If you are willing to go further as to [install `web-ext` for firefox](https://github.com/mozilla/web-ext/), make a git clone and try running in direcory:
```
$ web-ext run
```
See if the extension appears now. Then try opening the console window (as described above) and see the output.

**Comment by xGSTx** [comment]
Sorry, I guess i am done with my abilities here.

the curl thing worked in bash but not in firefox console:

`$ curl "http://xml.alexa.com/data?cli=10&dat=nsa&url=github.com"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     4  100     4    0     0     12      0 --:--:-- --:--:-- --:--:--    12Okay
`

I installed git and cloned web-ext, cd into the directory and ran web-ext run but it couldn't find the command.

I am just going to reinstall windows from scratch. if it is working for you in FF60+ well then it must work after a new reinstall of windows.

Sorry, i have no developer skills. i can't help on this level :(

**Comment by ntd** [comment]
I have the same problem: something seems to have changed on the Alexa side.

By leveraging [a free proxy](https://www.proxysite.com/), if I send [the request](https://xml.alexa.com/data?cli=10&dat=nsa&url=github.com) from an EU zone I get `Okay` as response body while sending from an US zone gives the proper XML response.

**Comment by xGSTx** [comment]
Oh wow!

Thanx for the input ntd. I was too dumb to locate the issue. haha. I also live in the EU hence the problem i guess. So, there you go. Now we know a lot more about the issue.

My Life is not worth living without this addon. Please fix this. Thanks.

**Comment by hiddenchamp** [comment]
I have the exact same problem. After upgrading to firefox 60 the alexa plugin got greyed out as show in the screenshot posted above by another user. This is the debug error I get:

> Connecting to localhost:53317, ws: false
> [Exception... "Component returned failure code: 0x80470002 (NS_BASE_STREAM_CLOSED) [nsIInputStream.available]"  nsresult: "0x80470002 (NS_BASE_STREAM_CLOSED)"  location: "JS frame :: resource://devtools/shared/base-loader.js -> resource://devtools/shared/security/socket.js :: onInputStreamReady :: line 328"  data: no]
> onInputStreamReady@resource://devtools/shared/base-loader.js -> resource://devtools/shared/security/socket.js:328:9

**Comment by jackdorsey** [comment]
It seems it does not work for certain IPs:
https://stackoverflow.com/questions/50279057/alexa-site-rank-api

It may be completely banned in the future so maybe it is better to parse this:
https://www.alexa.com/minisiteinfo/stackoverflow.com
or this:
https://www.alexa.com/siteinfo/stackoverflow.com

**Comment by hiddenchamp** [comment]
@Jack you are right.
I tried it with VPN and it worked for a few seconds. I have no idea what happened but after a minute or so it stopped working again.

Now whatever country I select it doesnt work.
BTW I am from Romania.

**Comment by fuyu** [comment]
@hiddenchamp Does the URLs provided by jackdorsey work for you?
```
It may be completely banned in the future so maybe it is better to parse this:
https://www.alexa.com/minisiteinfo/stackoverflow.com
or this:
https://www.alexa.com/siteinfo/stackoverflow.com
```

**Comment by hiddenchamp** [comment]
@fuyu yes those links work fine. I actually use them daily now that the plugin does not work.

**Comment by fuyu** [comment]
This should be fixed in v0.9.2: https://github.com/fuyu/alexa-rank-webextension/releases/tag/v0.9.2
